src/Microsoft.Diagnostics.Monitoring.StartupHook/Exceptions/GuardedExceptionSourceBase.cs

```diff
@@ -3,28 +3,16 @@
 
 using System;
 using System.Diagnostics;
-using System.Runtime.ExceptionServices;
 using System.Threading;
 
 namespace Microsoft.Diagnostics.Monitoring.StartupHook.Exceptions
 {
-    /// <summary>
-    /// Produces first chance exceptions from the current app domain.
-    /// </summary>
-    internal sealed class CurrentAppDomainExceptionSource :
-        ExceptionSourceBase,
-        IDisposable
+    internal abstract class GuardedExceptionSourceBase :
+        ExceptionSourceBase
     {
         private readonly ThreadLocal<bool> _handlingException = new();
 
-        private long _disposedState;
-
-        public CurrentAppDomainExceptionSource()
-        {
-            AppDomain.CurrentDomain.FirstChanceException += CurrentDomain_FirstChanceException;
-        }
-
-        private void CurrentDomain_FirstChanceException(object? sender, FirstChanceExceptionEventArgs e)
+        protected void RaiseExceptionGuarded(Exception exception)
         {
             DateTime timestamp = DateTime.UtcNow;
```

Comment:
nit: It is more common to implement IDisposable in the base class and Dispose(bool disposing) in derived classes.

Reply:
I think you wanted to leave this comment somewhere. In any event, I largely disagree with that approach in general because you shouldn't force concrete implementations to have the disposal pattern if there isn't anything to dispose. I also largely leave that out of any interface derivations because that consumer of the interface in most cases should not have knowledge of the disposability of an implementation nor be responsible for it, unless you are making it part of the contract that something that acquires that instance is now responsible for the lifetime of it.